### PR TITLE
chore: Fix autocomplete bugs

### PR DESCRIPTION
**Pull request by shamantraghav**
Closes : 

[DI-754](https://issues.corp.twilio.com/browse/DII-5)
[DII-5](https://issues.corp.twilio.com/browse/DII-5)

Screenshot for [DI-754](https://issues.corp.twilio.com/browse/DII-5):
Screenshot 2021-07-29 at 12 51 45 PM

When plugins:update is run: 
Screenshot 2021-07-29 at 12 59 23 PM

When any other command is run: 
Screenshot 2021-07-29 at 1 00 59 PM

Screenshot for [DII-5](https://issues.corp.twilio.com/browse/DII-5):

When autocomplete is run without the shell:
Screenshot 2021-07-29 at 1 06 33 PM

When autocomplete is run with the shell:
Screenshot 2021-07-29 at 1 08 16 PM

# Fixes #

A short description of what this PR does.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: src/hooks/postrun/plugin-postrun.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+const AUTOCOMLETE_WARNING = `If you’re using autocomplete, you’ll need to run ${chalk.bold(
+  'twilio autocomplete',
+)} after installing a plugin and then open a new terminal window. The CLI needs to re-build its cache.`;
+const AUTOCOMLETE_ALERT = `If you are running bash or zsh on macOS or Linux, you can run one of the two commands below (as appropriate for the shell you are using): ${chalk.bold(
+  'twilio autocomplete bash',
+)} or ${chalk.bold('twilio autocomplete zsh')}`;
+
+module.exports = async function pluginPostRun(options) {
+  // console.log(options);
```
Comment (onuzbee) [inline src/hooks/postrun/plugin-postrun.js:13]: Can be removed?

File: package.json [hunk 2]
```diff
@@ -111,6 +111,9 @@
       "plugins:preinstall": [
         "./src/hooks/plugin-install"
       ],
+      "postrun": [
```
Comment (shamantraghav) [inline package.json:114]: For info on how postrun event functions: https://oclif.io/docs/hooks

Comment (shamantraghav) [inline package.json:114]: For info on how postrun event funcctions: https://oclif.io/docs/hooks

File: src/services/post-install-utility.js [hunk 3]
```diff
@@ -48,11 +52,13 @@ class PostInstallDisplayManager {
 
     if (this.hasProjects()) {
       console.warn(chalk.yellowBright(` » ${PORT_WARNING}`));
+      console.warn(chalk.yellowBright(` » ${AUTOCOMLETE_WARNING}`));
```
Comment (onuzbee) [inline src/services/post-install-utility.js:55]: Should we not display autocomplete warning for a fresh install with no profiles configured as well? I might be out of context here, but I think it shouldn't be check based as what I could see from the ask, and should display this autocomplete warning for every install.

File: src/hooks/postrun/plugin-postrun.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable no-console */
+const { logger } = require('@twilio/cli-core').services.logging;
```
Comment (onuzbee) [inline src/hooks/postrun/plugin-postrun.js:2]: Not required

File: src/hooks/postrun/plugin-postrun.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+const AUTOCOMLETE_WARNING = `If you’re using autocomplete, you’ll need to run ${chalk.bold(
+  'twilio autocomplete',
+)} after installing a plugin and then open a new terminal window. The CLI needs to re-build its cache.`;
+const AUTOCOMLETE_ALERT = `If you are running bash or zsh on macOS or Linux, you can run one of the two commands below (as appropriate for the shell you are using): ${chalk.bold(
+  'twilio autocomplete bash',
+)} or ${chalk.bold('twilio autocomplete zsh')}`;
+
+module.exports = async function pluginPostRun(options) {
+  // console.log(options);
+  if (options.Command.id === 'plugins:update') {
```
Comment (shamantraghav) [inline src/hooks/postrun/plugin-postrun.js:14]: This is the custom event triggered.

- Once postrun is triggered, we check if the "plugins:update" command is run and display the necessary warnings.
- Similarly we do a check for autocomplete.

File: src/services/post-install-utility.js [hunk 6]
```diff
@@ -7,6 +7,10 @@ const PORT_WARNING = `Profiles exist with API keys in system keychain. Please ru
   'twilio profiles:port',
 )} to port all keys to the config file`;
 
+const AUTOCOMLETE_WARNING = `If you’re using autocomplete, you’ll need to run ${chalk.bold(
```
Comment (shamantraghav) [inline src/services/post-install-utility.js:10]: Adding the autocomplete warning for when the cli is updated

File: src/hooks/postrun/plugin-postrun.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable no-console */
+const { logger } = require('@twilio/cli-core').services.logging;
+const chalk = require('chalk');
+
+const AUTOCOMLETE_WARNING = `If you’re using autocomplete, you’ll need to run ${chalk.bold(
+  'twilio autocomplete',
+)} after installing a plugin and then open a new terminal window. The CLI needs to re-build its cache.`;
+const AUTOCOMLETE_ALERT = `If you are running bash or zsh on macOS or Linux, you can run one of the two commands below (as appropriate for the shell you are using): ${chalk.bold(
```
Comment (kridai) [inline src/hooks/postrun/plugin-postrun.js:8]: nit: the statement says `you can run one of the two commands below (as appropriate for the shell you are using)` but they are shown in the same line (as per the screenshot you shared), maybe we can add `\n` to have the commands show up in the next line.

File: src/hooks/postrun/plugin-postrun.js (hunk 4 above)
Comment (shamantraghav) [reply src/hooks/postrun/plugin-postrun.js]: Removed

File: src/hooks/postrun/plugin-postrun.js (hunk 7 above)
Comment (shamantraghav) [reply src/hooks/postrun/plugin-postrun.js]: Formatted it

File: src/hooks/postrun/plugin-postrun.js (hunk 1 above)
Comment (shamantraghav) [reply src/hooks/postrun/plugin-postrun.js]: Done

File: src/services/post-install-utility.js (hunk 3 above)
Comment (shamantraghav) [reply src/services/post-install-utility.js]: Nice spot! Made the change

**Review by Sindhura3 — APPROVED** [review]
Well done on the first PR! LGTM

**Review by shrutiburman — APPROVED** [review]
Looks good! 🚀

**Review by onuzbee — APPROVED** [review]
🚀  Good work!

**Review by kridai — APPROVED** [review]
👍

**Pull request merged**